sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,259 @@
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)
+            && classField.isAnnotationPresent(SearchableFieldProperty.class)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                    String.format("@SimpleFieldProperty and @SearchableFieldProperty cannot be present simultaneously "
+                        + "for %s", classField.getName())));
+        }
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)) {
+            SimpleFieldProperty simpleFieldPropertyAnnotation =
+                classField.getDeclaredAnnotation(SimpleFieldProperty.class);
+            searchField.setSearchable(false)
+                .setSortable(simpleFieldPropertyAnnotation.isSortable())
+                .setFilterable(simpleFieldPropertyAnnotation.isFilterable())
+                .setFacetable(simpleFieldPropertyAnnotation.isFacetable())
+                .setKey(simpleFieldPropertyAnnotation.isKey())
+                .setHidden(simpleFieldPropertyAnnotation.isHidden());
+        } else if (classField.isAnnotationPresent(SearchableFieldProperty.class)) {
+            if (!searchField.getType().equals(DataType.EDM_STRING)
+                && !searchField.getType().equals(DataType.collection(DataType.EDM_STRING))) {
+                throw logger.logExceptionAsError(new RuntimeException(String.format("SearchFieldProperty can only"
+                    + " be used on string properties. Property %s returns a %s value.",
+                    classField.getName(), searchField.getType())));
+            }
+            SearchableFieldProperty searchableFieldPropertyAnnotation =
+                classField.getDeclaredAnnotation(SearchableFieldProperty.class);
+            searchField.setSearchable(true)
+                .setSortable(searchableFieldPropertyAnnotation.isSortable())
+                .setFilterable(searchableFieldPropertyAnnotation.isFilterable())
+                .setFacetable(searchableFieldPropertyAnnotation.isFacetable())
+                .setKey(searchableFieldPropertyAnnotation.isKey())
+                .setHidden(searchableFieldPropertyAnnotation.isHidden());
+            if (!searchableFieldPropertyAnnotation.analyzer().isEmpty()) {
+                searchField.setAnalyzer(AnalyzerName.fromString((searchableFieldPropertyAnnotation.analyzer())));
+            }
+            if (!searchableFieldPropertyAnnotation.searchAnalyzer().isEmpty()) {
+                searchField.setAnalyzer(AnalyzerName.fromString((searchableFieldPropertyAnnotation.searchAnalyzer())));
+            }
+            if (!searchableFieldPropertyAnnotation.indexAnalyzer().isEmpty()) {
+                searchField.setAnalyzer(AnalyzerName.fromString((searchableFieldPropertyAnnotation.indexAnalyzer())));
+            }
+            if (searchableFieldPropertyAnnotation.synonymMaps().length != 0) {
+                searchField.setSynonymMaps(Arrays.asList(searchableFieldPropertyAnnotation.synonymMaps()));
```

Comment:
Should we attempt to cleanse any potential `null` values in the annotation value? Ignore this if Java doesn't allow `{ "aSynonym", null, "anotherSynonym" }`.

Reply:
@brjohnstmsft I think this is valid. 
Do you think we need extra step of removing null out of map or throw error here?

Reply:
@sima-zhu I don't recall if the service ignores or rejects null in the list of synonym maps. Nulls are no meaningful in that context, so I think it would be fine to ignore them.

Reply:
Will remove null from the list.